lib/tests/unit/modules/command_test.cc

```diff
@@ -0,0 +1,99 @@
+#include <catch.hpp>
+#include "root_path.hpp"
+#include "../../common/content_format.hpp"
+
+#include <boost/algorithm/string.hpp>
+
+#include <leatherman/execution/execution.hpp>
+#include <leatherman/util/scope_exit.hpp>
+
+#include <pxp-agent/configuration.hpp>
+#include <pxp-agent/modules/command.hpp>
+
+using namespace PXPAgent;
+
+namespace fs = boost::filesystem;
+namespace lth_exe = leatherman::execution;
+namespace lth_jc = leatherman::json_container;
+
+static const std::string SPOOL_DIR { std::string { PXP_AGENT_ROOT_PATH }
+                                     + "/lib/tests/resources/test_spool" };
+static const auto STORAGE = std::make_shared<ResultsStorage>(SPOOL_DIR, "0d");
+
+#ifdef _WIN32
+static const std::string ECHO_PARAMS { "{ \"command\": \"cmd.exe /c echo hello\" }"};
+#else
+static const std::string ECHO_PARAMS { "{ \"command\": \"echo hello\" }" };
+#endif
+
+static const std::string ECHO_COMMAND_TEXT {
+        (NON_BLOCKING_DATA_FORMAT % "\"0987\""
+         % "\"command\""
+         % "\"run\""
+         % ECHO_PARAMS
+         % "false").str() };
+
+static const PCPClient::ParsedChunks ECHO_COMMAND_CONTENT {
+        lth_jc::JsonContainer(ENVELOPE_TXT),
+        lth_jc::JsonContainer(ECHO_COMMAND_TEXT),
+        {},
+        0 };
+
+static void configureTest() {
+    if (!fs::exists(SPOOL_DIR) && !fs::create_directories(SPOOL_DIR)) {
+        FAIL("Failed to create the results directory");
+    }
+    Configuration::Instance().initialize(
+            [](std::vector<std::string>) {
+                return EXIT_SUCCESS;
+            });
+}
+
+static void resetTest() {
+    if (fs::exists(SPOOL_DIR)) {
+        fs::remove_all(SPOOL_DIR);
+    }
+}
+
+TEST_CASE("Modules::Command", "[modules]") {
+    SECTION("can successfully instantiate") {
+        REQUIRE_NOTHROW(Modules::Command(PXP_AGENT_BIN_PATH, STORAGE));
+    }
+}
+
+TEST_CASE("Modules::Command::hasAction", "[modules]") {
+    Modules::Command mod { PXP_AGENT_BIN_PATH, STORAGE };
+
+    SECTION("correctly reports false for a nonexistent action") {
+        REQUIRE_FALSE(mod.hasAction("foo"));
+    }
+
+    SECTION("correctly reports true for a real action") {
+        REQUIRE(mod.hasAction("run"));
+    }
+}
+
+TEST_CASE("Modules::Command::callAction", "[modules]") {
+    configureTest();
+    lth_util::scope_exit config_cleaner { resetTest };
+    Modules::Command mod { PXP_AGENT_BIN_PATH, STORAGE };
+
+    ActionRequest request { RequestType::NonBlocking, ECHO_COMMAND_CONTENT };
+    fs::path spool_path { SPOOL_DIR };
+    auto results_dir = (spool_path / request.transactionId()).string();
+    fs::create_directories(results_dir);
+    request.setResultsDir(results_dir);
+
+    auto response = mod.executeAction(request);
+
+    SECTION("executes a command") {
```

Comment:
This is a successful execution; would it make sense to have ones that failed in various ways to sure the command validation mechanisms all worked?